### DISCUSSION TITLE
How to auto discover package components

### DISCUSSION THREAD

**Discussion by bilogic** [livewire/livewire#3434] [1. Help] [answered]
I'm aware of the need to register my package's components in my package service provider.
```PHP
Livewire::component('header', HeaderComponent::class);
```
Just curious if there is a way to do this automatically? Thanks.

**Comment by joshhanley** [comment, marked as answer]
@bilogic there isn't a way to do it automatically.

You could look at Livewire's service provider and see how it autoloads components and copy the code to your service provider to do the same.

Hope this helps!

**Comment by bilogic** [comment]
https://github.com/livewire/livewire/blob/520964a3e70cf58692fe2a5ac30572bebdd710f4/src/LivewireServiceProvider.php#L93-L113

Thanks @joshhanley, found it!

**Reply by joshhanley** [reply]
@bilogic no worries! 🙂
